src/AzureAuth/CommandMain.cs

```diff
@@ -167,6 +169,23 @@ public Alias TokenFetcherOptions
 
         private AuthMode CombinedAuthMode => this.AuthModes.Aggregate((a1, a2) => a1 | a2);
 
+        /// <summary>
+        /// Get the combination of the prefix <see cref="PromptHintPrefix"/> with user-supplied option.
+        /// By default "AzureAuth" is the prefix.
+        /// </summary>
+        /// <param name="promptHintOption">The prompt hint provided by users.</param>
+        /// <returns>the combined value.</returns>
+        public static string GetActualPromptHint(string promptHintOption)
+        {
+            string promptHint = PromptHintPrefix;
+            if (!string.IsNullOrEmpty(promptHintOption))
+            {
+                promptHint = $"{PromptHintPrefix}: {promptHintOption}";
+            }
+
+            return promptHint;

```

Comment:
This is a case where I would support also using multiple return because of how contained the logic is. Also I think it's generally easier for readers if you avoid using `!` if you can. Consider:
```suggestion
            if (string.IsNullOrEmpty(promptHintOption))
            {
                return PromptHintPrefix;
            }
            return $"{PromptHintPrefix}: {promptHintOption}";
```